### DISCUSSION TITLE
New User Role is not working

### DISCUSSION THREAD

**Discussion by shubh-mehrotra** [nasirkhan/laravel-starter#430] [Q&A]
I have created a new role and created new routes with middlewares (`'middleware' => ['web', 'auth', 'can:manage_purchaseorders']`) but new user isn't able to access those routes.

Need your suggestions here.

**Comment by nasirkhan** [comment]
Did you follow this doc? can you please check again there to understand how it works https://github.com/nasirkhan/laravel-starter/wiki/Role-Permission

**Reply by shubh-mehrotra** [reply]
I followed the same steps but the new role's user is not able to access the routes -
https://prnt.sc/kaMHSfhZK_iK
https://prnt.sc/oZtECrkMTEwm
https://prnt.sc/3H9xQYUiwr2J
https://prnt.sc/IcUXQtYNoFg9
https://prnt.sc/PR5uLFu6z4A0
https://prnt.sc/aUDGhepvXNAV

**Reply by nasirkhan** [reply]
I created a new permission and assigned that to a role. Then I used that permission on a route and from a controller and view files. All the tests were successful and permission is working fine. 

It may be a case that the permissions/roles are cached and not updated when you are checking the routes. You may clear this cache by using the following command. 

```
php artisan cache:forget spatie.permission.cache
```

**Reply by shubh-mehrotra** [reply]
It seems the problem is with the permission name, if I am renaming it to `view_purchaseorders` instead of `manage_purchaseorders`, it's working fine.

not exactly sure about the reason behind it, @nasirkhan are you aware about it?
